src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java

```diff
@@ -0,0 +1,50 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import org.bukkit.entity.Player;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class ExprPlayerProtocolVersion extends SimplePropertyExpression<Player, Integer> {
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.network.NetworkClient")) {
+			register(ExprPlayerProtocolVersion.class, Integer.class, "[the] [client] protocol version", "players");
+		}
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Player player) {
+		return player.getProtocolVersion();
```

Comment:
The documentation says that this method returns `-1` when it is unknown, maybe it'd be a better idea for Skript to make it return null?

Reply:
I agree with you on that.